### PR TITLE
CDF Login changes

### PR DESCRIPTION
**Pull request by Vipinofficial11**
Adding functionality to login on CDF using OTA credentials.

### CODE REVIEW

File: src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java [hunk 1]
```diff
@@ -105,11 +115,20 @@ public static void waitForPipelineToTransitionToStatus(String pipelineStatus, lo
...
    * Timeout: {@link ConstantsUtil#PIPELINE_RUN_TIMEOUT_SECONDS}
    */
-  public static void waitTillPipelineRunCompletes() {
+  public static void waitTillPipelineRunCompletes() throws InterruptedException, IOException {
+    // Adding a page refresh in case tests are running on CDF to update the pipeline status.
+    if (Boolean.parseBoolean(SeleniumHelper.readParameters(ConstantsUtil.TESTONCDF)) ||
+      Boolean.parseBoolean(SeleniumHelper.readParameters(ConstantsUtil.TESTONHDF))) {
+      do {
+        PageHelper.refreshCurrentPage();
+        Thread.sleep(1000 * 60);
```
Comment (poojantcs) [inline src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java]: Have we tried using any dynamic waits here?

File: src/main/java/io/cdap/e2e/pages/actions/CdfSignInActions.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        cdfSignInLocator = SeleniumHelper.getPropertiesLocators(CdfSignInLocator.class);
+    }
+
+    public static void login() throws IOException {
+        ElementHelper.sendKeys(cdfSignInLocator.cdfUsername, SeleniumHelper.readParameters(CDFUSERNAME));
+        ElementHelper.clickOnElement(cdfSignInLocator.nextButton);
+        ElementHelper.sendKeys(cdfSignInLocator.cdfPassword, SeleniumHelper.readParameters(CDFPASSWORD));
+        ElementHelper.clickOnElement(cdfSignInLocator.nextButton);
+
+        // Adding optional operations here.
```
Comment (poojantcs) [inline src/main/java/io/cdap/e2e/pages/actions/CdfSignInActions.java]: More context on why we need to add optional steps would help

File: src/main/java/io/cdap/e2e/pages/actions/CdfSignInActions.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          cdfSignInLocator.clickOnAllowButton());
+
+        ElementHelper.clickIfDisplayed(cdfSignInLocator.clickOnContinueButton(), ConstantsUtil.MEDIUM_TIMEOUT_SECONDS,
+          cdfSignInLocator.locatePluginNameInList(ConstantsUtil.FIRST_PLUGIN_IN_LIST, "Source"));
+
+        ElementHelper.clickIfDisplayed(cdfSignInLocator.clickOnAllowButton(), ConstantsUtil.MEDIUM_TIMEOUT_SECONDS,
+          cdfSignInLocator.locatePluginNameInList(ConstantsUtil.FIRST_PLUGIN_IN_LIST, "Source"));
+    }
+
+    public static boolean logged() {
```
Comment (poojantcs) [inline src/main/java/io/cdap/e2e/pages/actions/CdfSignInActions.java]: Rename: isUserLoggedInCDF

File: src/main/java/io/cdap/e2e/pages/locators/CdfSignInLocator.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    @FindBy(how = How.XPATH, using = "//*[contains(text(),'Next')]//parent::button")
+    public WebElement nextButton;
+
+    public static By locatorOfEmailTextBox() {
+        return By.xpath("//input[@name='identifier']");
+    }
+
+    public By selectTestAccount() {
+        return By.xpath("//*[contains(text(),'CloudDataFution Automation')]");
```
Comment (poojantcs) [inline src/main/java/io/cdap/e2e/pages/locators/CdfSignInLocator.java]: Create a dynamic locator. Read the Test Account name from the properties file.

File: src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java (hunk 1 above)
Comment (Vipinofficial11) [reply src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java]: Didn't tried before, used now. (https://github.com/cloudsufi/cdap-e2e-tests/pull/2/commits/2fa8b80cc365818abeaf5bec5c89ab1185aa83dc#diff-cfa45894e4592a5d6ed32f485ed2594ae01b9a3b154e60820a599164e0b52e80R128)

File: src/main/java/io/cdap/e2e/pages/actions/CdfSignInActions.java (hunk 2 above)
Comment (Vipinofficial11) [reply src/main/java/io/cdap/e2e/pages/actions/CdfSignInActions.java]: Added [2fa8b80cc365818abeaf5bec5c89ab1185aa83dc]

File: src/main/java/io/cdap/e2e/pages/actions/CdfSignInActions.java (hunk 3 above)
Comment (Vipinofficial11) [reply src/main/java/io/cdap/e2e/pages/actions/CdfSignInActions.java]: Updated.

File: src/main/java/io/cdap/e2e/pages/locators/CdfSignInLocator.java (hunk 4 above)
Comment (Vipinofficial11) [reply src/main/java/io/cdap/e2e/pages/locators/CdfSignInLocator.java]: Updated (https://github.com/cloudsufi/cdap-e2e-tests/pull/2/commits/2fa8b80cc365818abeaf5bec5c89ab1185aa83dc#diff-b3632a67370578b810c74200a2a8673e2a0f17fefb10874d9bf696b660f6ca0aR17)

**Pull request closed without merging**